### PR TITLE
Issue112 rqmultipart tests

### PR DESCRIPTION
**Pull request by ekondrashev**
Please review this pull request for the 112 issue
https://github.com/yegor256/takes/issues/112

The RqMultipartTest was missing some edge tests to cover RqMultipart 100%. I've added number of negative tests along with few missing positive tests.

### CODE REVIEW

**Comment by davvd** [comment]
Let me find a reviewer for this pull request, thanks for submitting it

**Comment by davvd** [comment]
@ggajos review this one, please

**Comment by ggajos** [comment]
@davvd please include some brief PR description, how/why you're solving issue

**Comment by ggajos** [comment]
@davvd please fix qulice issues

File: src/test/java/org/takes/rq/RqMultipartTest.java [hunk 1]
```diff
@@ -42,6 +44,129 @@
...
     /**
+     * Carriage return constant.
+     */
+    private static final String CR = "\r\n";
+    /**
+     * Two RqMultipart objects are equal when inited with equal Request objects.
+     * @throws IOException if some problem inside
+     */
+    @Test
+    public void equalsReturnsTrue() throws IOException {
```
Comment (ggajos) [inline src/test/java/org/takes/rq/RqMultipartTest.java]: @ekondrashev see http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names for test name and test description.

File: src/test/java/org/takes/rq/RqMultipartTest.java [hunk 2]
```diff
@@ -42,6 +44,129 @@
...
+                "",
+                "\r\t\n\u20ac\n\n\n\t\r\t\n\n\n\r\nthe end ",
+                "--AaB01x--"
+            )
+        );
+        Assert.assertEquals(new RqMultipart(req), new RqMultipart(req));
+    }
+
+    /**
+     * RqMultipart throws IOException on missing closing boundary.
```
Comment (ggajos) [inline src/test/java/org/takes/rq/RqMultipartTest.java]: @ekondrashev see http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names for test name and test description.

File: src/test/java/org/takes/rq/RqMultipartTest.java [hunk 3]
```diff
@@ -42,6 +44,129 @@
...
+                "",
+                "447 N Wolfe Rd, Sunnyvale, CA 94085",
+                "Content-Transfer-Encoding: uwf-8"
+            )
+        );
+        new RqMultipart(req);
+    }
+
+    /**
+     * RqMultipart throws IOException on missing name
```
Comment (ggajos) [inline src/test/java/org/takes/rq/RqMultipartTest.java]: @ekondrashev see http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names for test name and test description.

File: src/test/java/org/takes/rq/RqMultipartTest.java [hunk 4]
```diff
@@ -42,6 +44,129 @@
...
+                "",
+                "\r\t\n\u20ac\n\n\n\t\r\t\n\n\n\r\nthe  end",
+                "--AaB00x--"
+            )
+        );
+        new RqMultipart(req);
+    }
+
+    /**
+     * RqMultipart throws IOException on missing boundary Content-Type header.
```
Comment (ggajos) [inline src/test/java/org/takes/rq/RqMultipartTest.java]: @ekondrashev `can throw` / see http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names for test name and test description.

File: src/test/java/org/takes/rq/RqMultipartTest.java [hunk 5]
```diff
@@ -42,6 +44,129 @@
...
+                "--AaB01x",
+                // @checkstyle LineLength (1 line)
+                "Content-Disposition: form-data; name=\"data\"; filename=\"a.bin\"",
+                "Content-Transfer-Encoding: uyf-8",
+                "",
+                "\r\t\n\u20ac\n\n\n\t\r\t\n\n\n\r\nthe end ",
+                "--AaB01x--"
+            )
+        );
+        Assert.assertEquals(new RqMultipart(req), new RqMultipart(req));
```
Comment (ggajos) [inline src/test/java/org/takes/rq/RqMultipartTest.java]: @ekondrashev always use MatcherAssert

**Comment by ggajos** [comment]
@ekondrashev your PR is very big, please try to submit smaller ones in future

**Comment by ggajos** [comment]
@ekondrashev you're creating similar requests over and over, maybe it would be good to extract method that will help you with creation of requests?

File: src/test/java/org/takes/rq/RqMultipartTest.java (hunk 1 above)
Comment (ekondrashev) [inline src/test/java/org/takes/rq/RqMultipartTest.java]: Thanks for the link.

But after reading I can't see what would be correct name using "RqMultipart can .." pattern:
"RqMultipart can equals to the same data inited RqMultipart" as for the doc and
equalsToTheSameDataInitedRqMultipart as for method name.

Not sure it sounds good enough.
Could you please provide your ideas?

**Comment by ggajos** [comment]
@ekondrashev one comment is gone but regarding naming you can say `satisfiesEqualsContract` instead of `equalsReturnsTrue`. It's more abstract, but your current changes are ok.

File: src/test/java/org/takes/rq/RqMultipartTest.java [hunk 6]
```diff
@@ -42,32 +45,123 @@
...
-    public void parsesHttpBody() throws IOException {
+    public void returnsTrueOnEqualsCall() throws IOException {
+        final Request req = create(
+            Joiner.on(CR).join(
+                "Content-Disposition: form-data; name=\"addres\"",
+                "",
+                "449 N Wolfe Rd, Sunnyvale, CA 94085"
+            ),
+                // @checkstyle LineLength (1 line)
+                "Content-Disposition: form-data; name=\"data\"; filename=\"a.bin\""
```
Comment (ggajos) [inline src/test/java/org/takes/rq/RqMultipartTest.java]: @ekondrashev let's use new joiner here on `; ` character and avoid line length checkstyle suppression

File: src/test/java/org/takes/rq/RqMultipartTest.java [hunk 7]
```diff
@@ -85,4 +179,75 @@ public void parsesHttpBody() throws IOException {
...
+     */
+    @Test
+    public void returnsEmptyIteratorOnInvalidPartRequest() throws IOException {
+        final Request req = create(
+            Joiner.on(CR).join(
+                "Content-Disposition: form-data; name=\"address\"",
+                "",
+                "443 N Wolfe Rd, Sunnyvale, CA 94085",
+                // @checkstyle LineLength (1 line)
+                "Content-Disposition: form-data; name=\"data\"; filename=\"a.zip\""
```
Comment (ggajos) [inline src/test/java/org/takes/rq/RqMultipartTest.java]: @ekondrashev let's use new joiner here on `; ` character and avoid line length checkstyle suppression

**Comment by ggajos** [comment]
@ekondrashev few minor comments above

**Comment by ggajos** [comment]
@rultor merge please

**Comment by yegor256** [comment]
@rultor merge

**Comment by davvd** [comment]
@elenavolokhova please, review this task for compliance with our [quality rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by elenavolokhova** [comment]
@ggajos One comment regarding addressing the comments in the ticket:
>davvd commented 12 days ago
Let me find a reviewer for this pull request, thanks for submitting it

Could you ask to fix them in future tickets?

**Comment by ggajos** [comment]
@davvd please address you're comments directly

**Comment by ggajos** [comment]
@elenavolokhova sure, will do, thank you for feedback

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks for the review, we'll try better next time

**Comment by davvd** [comment]
@ggajos just added 10 mins to @elenavolokhova (for QA), payment ID is `55563732`; paid, thanks, added **24 mins** to your account, payment `AP-2SB34765BR049391P`, 143 hours and 48 mins was spent; you're getting extra minutes here (c=9); +24 added to your rating, at the moment it is: [+3724](http://www.netbout.com/b/35094?open=rating)

**Pull request merged**